src/legacy/core_plugins/vis_type_timeseries/server/plugin.ts

```diff
@@ -38,16 +40,26 @@ export interface CustomCoreSetup {
 
 export class MetricsServerPlugin {
   public initializerContext: PluginInitializerContext;
+  private validationTelementryService: ValidationTelemetryService;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.initializerContext = initializerContext;
+    this.validationTelementryService = new ValidationTelemetryService();
   }
 
-  public setup(core: CoreSetup & CustomCoreSetup) {
+  public async setup(
+    core: CoreSetup & CustomCoreSetup,
+    deps: { usageCollection?: UsageCollectionSetup }
+  ) {
     const { http } = core;
 
+    const validationTelemetrySetup = await this.validationTelementryService.setup(core, {
+      ...deps,
+      kibanaIndex: http.server.config().get('kibana.index'),
```

Comment:
Not sure that we can pass `kibanaIndex` in this way. If I'm not wrong for all setup methods we should use the following convension: 
`setup: (core: Core Setup, plugins: {dependent Plugins}) => { public API}`

My suggestion: you can  get `kibana.index` directly in `validationTelementryService `

Reply:
That's right, I will change to this setup when merging in master that switches to the new core apis